File: app/src/main/java/com/ahmadabuhasan/appgithubuser/ui/FollowingFragment.java
package com.ahmadabuhasan.appgithubuser.ui;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.ahmadabuhasan.appgithubuser.adapter.FollowAdapter;
import com.ahmadabuhasan.appgithubuser.databinding.FragmentFollowingBinding;
import com.ahmadabuhasan.appgithubuser.viewmodel.UserViewModel;

import org.jetbrains.annotations.NotNull;

import es.dmoral.toasty.Toasty;

public class FollowingFragment extends Fragment {

    FragmentFollowingBinding binding;
    private UserViewModel followingViewModel;
    private FollowAdapter followAdapter;

    public FollowingFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(@NotNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding = FragmentFollowingBinding.inflate(inflater, container, false);
        View view = binding.getRoot();

        followAdapter = new FollowAdapter(getContext());
        followAdapter.notifyDataSetChanged();
        binding.rvFollowing.setLayoutManager(new LinearLayoutManager(getContext()));
        binding.rvFollowing.setAdapter(followAdapter);
        binding.rvFollowing.setHasFixedSize(true);

        followingViewModel = new ViewModelProvider(this, new ViewModelProvider.NewInstanceFactory()).get(UserViewModel.class);
        followingViewModel.setFollowing(UserDetailActivity.dataUser);
        followingViewModel.getFollowingUser().observe(getViewLifecycleOwner(), follows -> {
            if (follows.size() != 0) {
                followAdapter.setFollowList(follows);
            } else {
                Toasty.info(requireContext(), "Following Not Found", Toasty.LENGTH_SHORT).show();
            }
        });
        return view;
    }

    @Override
    public void onResume() {
        super.onResume();
        followingViewModel.setFollowers(UserDetailActivity.dataUser);
    }
}

File: Submission-Kotlin/app/src/main/java/com/ahmadabuhasan/appgithubuser/ui/FollowingFragment.kt
package com.ahmadabuhasan.appgithubuser.ui

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.recyclerview.widget.LinearLayoutManager
import com.ahmadabuhasan.appgithubuser.adapter.FollowAdapter
import com.ahmadabuhasan.appgithubuser.databinding.FragmentFollowingBinding
import com.ahmadabuhasan.appgithubuser.model.ResponseFollow
import com.ahmadabuhasan.appgithubuser.viewmodel.MainViewModel

class FollowingFragment : Fragment() {

    private val viewModel: MainViewModel by viewModels()
    private val adapter = FollowAdapter()

    private lateinit var binding: FragmentFollowingBinding
    private val _binding get() = binding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        // Inflate the layout for this fragment
        binding = FragmentFollowingBinding.inflate(inflater, container, false)
        return _binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        showViewModel()
        showRecyclerView()
        viewModel.getIsLoading.observe(viewLifecycleOwner, this::showLoading)
    }

    private fun showViewModel() {
        viewModel.following(UserDetailActivity.username)
        viewModel.getFollowing.observe(viewLifecycleOwner) { following ->
            if (following.size != 0) {
                adapter.setData(following)
            } else {
                Toast.makeText(context, "Following Not Found", Toast.LENGTH_SHORT).show()
            }
        }
    }

    private fun showRecyclerView() {
        with(binding.rvFollowing) {
            layoutManager = LinearLayoutManager(requireActivity())
            setHasFixedSize(true)
            adapter = adapter
        }

        adapter.setOnItemClickCallback { data -> selectedUser(data) }
    }

    private fun selectedUser(user: ResponseFollow) {
        Toast.makeText(context, "You choose ${user.login}", Toast.LENGTH_SHORT).show()

        val i = Intent(activity, UserDetailActivity::class.java)
        i.putExtra(UserDetailActivity.EXTRA_USER, user.login)
        startActivity(i)
    }

    private fun showLoading(isLoading: Boolean) =
        binding.progressBar.visibility == if (isLoading) View.VISIBLE else View.GONE

    override fun onResume() {
        super.onResume()
        viewModel.following(UserDetailActivity.username)
    }
}